### PR TITLE
Add ReorderChannels preprocessing layer

### LINKED ISSUE #15705: Reorder channel layer for smooth native RGB - BGR
**Here's why we have that policy:**.

Keras developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

**System information**.

- TensorFlow version (you are using): 2.7
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state**.

The feature is a `tf.keras` layer to let us reorder the channels of a specific dimension on during the feedforward operation without having to rely on custom implementation. This would be nice to finetune and deploy models with RGB and BGR without having to change the software that calls the TF Serving API. 

**Will this change the current api? How?**
```python
rgb2bgr_layer = tf.keras.layers.ReorderChannel(axis=-1, order[2, 1, 0])
```

**Who will benefit from this feature?**
Everyone who wants to try and deploy different models without changing the software that calls the TF Serving API. 

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**

- Do you want to contribute a PR? (yes/no): yes
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing): Create a ReorderChannel layer that receives parameters as "axis" and order. It will reorder the channels of the axis according to the axis. Simple as that.

### PR DESCRIPTION
**Pull request by piEsposito**
Closes #15705 .

Adds `tf.keras.layers.ReorderChannels` which let's the user permutate the channels of a tensor along a specific axis using `tf.gather` operation. Specially useful for in-model conversion from RGB to BGR. 

Also added two tests for images with channels first and channels last.

### CODE REVIEW

**Comment by piEsposito** [comment]
Please rescan for SLA, I've just signed it.

**Comment by LukeWood** [comment]
We actually want this on keras-cv.  If I remember correctly we didn't think this belonged in the core root keras repo.   https://github.com/keras-team/keras-cv/issues/20

Feel free to introduce this there.  Apologies if this wasn't clear.  

You can see https://github.com/keras-team/keras-cv/pull/22 for an example.  (I'm still working on unit tests there).  This repo is under construction so I'm slowly onboarding contributors even though it's not publicly documented yet... :)
